Sources/OutlineView/OutlineViewDragAndDrop.swift

```diff
@@ -0,0 +1,139 @@
+    ///
+    /// - Parameter item: The Pasteboard item that is being dragged into the OutlineView
+    ///
+    /// - Returns: nil if the Pasteboard item is not readable by this receiver, or
+    ///   a tuple with the decoded item and its associated PasteboardType, which must
+    ///   be included in `acceptedTypes`
+    func readPasteboard(item: NSPasteboardItem) -> DraggedItem<DataElement>?
+    
+    /// Called continuously as a dragged item is moved over the `OutlineView`, this is
+    /// used to define the behavior of the `OutlineView` and the drag cursor.
+    ///
+    /// - Parameter target: A `DropTarget` object describing where the dragged item
+    ///   is being dragged over.
+    ///
+    /// - Returns: A case of `ValidationResult`, which will either highlight the area
+    ///   of the `OutlineView` where the item will be dropped, or some other behavior.
+    ///   See `ValidationResult` for possible return values.
+    func validateDrop(target: DropTarget<DataElement>) -> ValidationResult<DataElement>
+    
+    /// Called once the mouse is released during a drag into the `OutlineView`, after
+    /// `validateDrop(target:)` has returned a case other than `deny`, this function should
+    /// handle updating the data source
+    ///
+    /// - Parameter target: A `DropTarget` item with all the information about
+    ///   where in the `OutlineView` heirarchy the new item will be dropped, and what
+    ///   the item(s) to be dropped is.
+    ///
+    /// - Returns: a boolean indicating that the drop was successful.
+    func acceptDrop(target: DropTarget<DataElement>) -> Bool
+}
+
+@available(macOS 10.15, *)
+public enum NoDropReceiver<Element: Identifiable>: DropReceiver {
+    public typealias DataElement = Element
+    
+    public func readPasteboard(item: NSPasteboardItem) -> DraggedItem<Element>? {
+        fatalError()
+    }
+    
+    public func validateDrop(target: DropTarget<Element>) -> ValidationResult<Element> {
+        fatalError()
+    }
+    
+    public func acceptDrop(target: DropTarget<Element>) -> Bool {
+        fatalError()
+    }
+}
+
+public typealias DragSourceWriter<D> = (D) -> NSPasteboardItem?
+public typealias DraggedItem<D> = (item: D, type: NSPasteboard.PasteboardType)
+
+/// An object describing what items are being dragged into an `OutlineView`, and
+/// where in the data heirarchy they are being dropped.
+@available(macOS 10.15, *)
+public struct DropTarget<D> {
+    /// A non-empty array of `DraggedItem` tuples, each with the item
+    /// that is being dragged, and the `NSPasteboard.PasteboardType` that
+    /// generated the item from the dragging pasteboard.
+    public var items: [DraggedItem<D>]
+    
+    /// The `OutlineView` data element into which the target is dropping
+    /// items.
+    ///
+    /// If nil, the items are intended to drop into the root of
+    /// the data heirarchy. Otherwise, they are to be dropped into the given
+    /// item's children array.
```

Comment:
```suggestion
    /// If `nil`, the items are intended to be dropped into the root of
    /// the data heirarchy. Otherwise, they are to be dropped into the given
    /// item's children.
```

Reply:
Got it https://github.com/Sameesunkaria/OutlineView/pull/10/commits/9a66eb8d6576eb16ca2d046bf17d013ac32249e3